File: frontend/services/sms.ts
/**
 * SMS Service for Frontend
 * Sends SMS notifications after registration and login
 * Non-blocking: if SMS fails, auth still succeeds
 */

const SMS_API_URL = process.env.NEXT_PUBLIC_API_BASE_URL ?? "http://localhost:8000";

/**
 * Call backend SMS endpoint
 */
async function sendSMSRequest(endpoint: string, payload: any): Promise<boolean> {
  try {
    const response = await fetch(`${SMS_API_URL}/api/sms${endpoint}`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(payload),
      cache: "no-store"
    });

    if (!response.ok) {
      console.warn(`SMS request failed: ${response.status}`);
      return false;
    }

    const data = await response.json();
    return data.success !== false;
  } catch (error) {
    console.error("SMS request error:", error);
    return false;
  }
}

/**
 * Send registration success SMS
 */
export async function sendRegistrationSMS(
  name: string,
  phone: string,
  role: "farmer" | "driver"
): Promise<boolean> {
  console.log(`Sending registration SMS to ${phone}...`);
  
  const success = await sendSMSRequest("/registration", {
    name,
    phone,
    role
  });

  if (success) {
    console.log("Registration SMS sent successfully");
  } else {
    console.warn("Registration SMS failed - but auth will continue");
  }

  return success;
}

/**
 * Send login alert SMS
 */
export async function sendLoginSMS(name: string, phone: string): Promise<boolean> {
  console.log(`Sending login SMS to ${phone}...`);
  
  const success = await sendSMSRequest("/login", {
    name,
    phone
  });

  if (success) {
    console.log("Login SMS sent successfully");
  } else {
    console.warn("Login SMS failed - but auth will continue");
  }

  return success;
}

/**
 * Send driver assignment SMS
 */
export async function sendDriverAssignmentSMS(
  driverName: string,
  phone: string,
  orderId: string
): Promise<boolean> {
  console.log(`Sending driver assignment SMS to ${phone}...`);
  
  return sendSMSRequest("/driver-assignment", {
    driver_name: driverName,
    phone,
    order_id: orderId
  });
}

/**
 * Send bundle created SMS
 */
export async function sendBundleCreatedSMS(
  farmerName: string,
  phone: string,
  bundleId: string,
  savings: number
): Promise<boolean> {
  console.log(`Sending bundle created SMS to ${phone}...`);
  
  return sendSMSRequest("/bundle-created", {
    farmer_name: farmerName,
    phone,
    bundle_id: bundleId,
    savings
  });
}

/**
 * Send trip update SMS
 */
export async function sendTripUpdateSMS(
  userName: string,
  phone: string,
  tripId: string,
  status: string
): Promise<boolean> {
  console.log(`Sending trip update SMS to ${phone}...`);
  
  return sendSMSRequest("/trip-update", {
    user_name: userName,
    phone,
    trip_id: tripId,
    status
  });
}

/**
 * Send generic SMS (for testing/custom messages)
 */
export async function sendCustomSMS(phone: string, message: string): Promise<boolean> {
  console.log(`Sending custom SMS to ${phone}...`);
  
  return sendSMSRequest("/custom", {
    phone,
    message
  });
}

File: backend/routes/sms.py
"""
SMS API Routes for KrishiBundle
Handles SMS notification endpoints
"""

from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from services.sms_service import (
    send_registration_sms,
    send_login_sms,
    send_driver_assignment_sms,
    send_bundle_created_sms,
    send_trip_update_sms
)

router = APIRouter()


# Request models
class RegistrationSMSRequest(BaseModel):
    name: str
    phone: str
    role: str


class LoginSMSRequest(BaseModel):
    name: str
    phone: str


class DriverAssignmentSMSRequest(BaseModel):
    driver_name: str
    phone: str
    order_id: str


class BundleCreatedSMSRequest(BaseModel):
    farmer_name: str
    phone: str
    bundle_id: str
    savings: int


class TripUpdateSMSRequest(BaseModel):
    user_name: str
    phone: str
    trip_id: str
    status: str


class CustomSMSRequest(BaseModel):
    phone: str
    message: str


# Response model
class SMSResponse(BaseModel):
    success: bool
    message: str


# Endpoints
@router.post("/registration", response_model=SMSResponse)
async def send_registration_notification(request: RegistrationSMSRequest):
    """Send registration success SMS"""
    try:
        success = send_registration_sms(request.name, request.phone, request.role)
        return SMSResponse(
            success=success,
            message="Registration SMS sent successfully" if success else "Failed to send registration SMS"
        )
    except Exception as e:
        print(f"Registration SMS error: {e}")
        # Don't throw error - SMS is non-critical
        return SMSResponse(success=False, message=str(e))


@router.post("/login", response_model=SMSResponse)
async def send_login_notification(request: LoginSMSRequest):
    """Send login alert SMS"""
    try:
        success = send_login_sms(request.name, request.phone)
        return SMSResponse(
            success=success,
            message="Login SMS sent successfully" if success else "Failed to send login SMS"
        )
    except Exception as e:
        print(f"Login SMS error: {e}")
        # Don't throw error - SMS is non-critical
        return SMSResponse(success=False, message=str(e))


@router.post("/driver-assignment", response_model=SMSResponse)
async def send_driver_assignment_notification(request: DriverAssignmentSMSRequest):
    """Send driver assignment SMS"""
    try:
        success = send_driver_assignment_sms(request.driver_name, request.phone, request.order_id)
        return SMSResponse(
            success=success,
            message="Driver assignment SMS sent successfully" if success else "Failed to send assignment SMS"
        )
    except Exception as e:
        print(f"Driver assignment SMS error: {e}")
        return SMSResponse(success=False, message=str(e))


@router.post("/bundle-created", response_model=SMSResponse)
async def send_bundle_created_notification(request: BundleCreatedSMSRequest):
    """Send bundle created SMS"""
    try:
        success = send_bundle_created_sms(
            request.farmer_name,
            request.phone,
            request.bundle_id,
            request.savings
        )
        return SMSResponse(
            success=success,
            message="Bundle SMS sent successfully" if success else "Failed to send bundle SMS"
        )
    except Exception as e:
        print(f"Bundle created SMS error: {e}")
        return SMSResponse(success=False, message=str(e))


@router.post("/trip-update", response_model=SMSResponse)
async def send_trip_update_notification(request: TripUpdateSMSRequest):
    """Send trip update SMS"""
    try:
        success = send_trip_update_sms(
            request.user_name,
            request.phone,
            request.trip_id,
            request.status
        )
        return SMSResponse(
            success=success,
            message="Trip update SMS sent successfully" if success else "Failed to send trip update SMS"
        )
    except Exception as e:
        print(f"Trip update SMS error: {e}")
        return SMSResponse(success=False, message=str(e))


@router.post("/custom", response_model=SMSResponse)
async def send_custom_notification(request: CustomSMSRequest):
    """Send custom SMS (for testing)"""
    try:
        from services.sms_service import sms_service
        success = sms_service.send_sms(request.phone, request.message, sms_type="custom")
        return SMSResponse(
            success=success,
            message="Custom SMS sent successfully" if success else "Failed to send custom SMS"
        )
    except Exception as e:
        print(f"Custom SMS error: {e}")
        return SMSResponse(success=False, message=str(e))
